### ISSUE TITLE
[App] `pip install lightning[app]` installs additional packages which may not be required

### ISSUE THREAD

**Issue by rohitgr7** [Lightning-AI/lightning#14716]
Following packages are installed too when installing lightning app using `pip install lightning[app]`.
I don't think it should install these. PL should be optional and torch should not be installed.

```console
lightning-lite==1.8.0.post1
tensorboard==2.10.0
torch==1.12.1
torchmetrics==0.9.3
tqdm==4.64.1
```

cc @carmocca @akihironitta @borda

**Comment by rohitgr7** [comment]
yes, it still installs the updated list

**Comment by Borda** [comment]
well, `lightning` does not have implemented any extras yet, see: #15542 and #15245

**Comment by awaelchli** [comment]
Closing the old issue, since the current package design will likely stay this way for a while.
```
pip install lightning
``` 
installs the pytorch/fabric dependencies, but no app, it's by design now.
```
pip install lightning[app]
``` 
installs the additional app dependencies. 

When only app is desired, we have the standalone package:
```
pip install lightning-app
```
